packages/ice-plugin-dll/lib/buildDll.js

```diff
@@ -37,36 +37,39 @@ async function generateDllVendorFile(outputDir, dependencies) {
   fs.writeFileSync(vendor, data);
 }
 
-async function includeDllInHTML(outputDir, defaultAppHtml, log) {
+async function includeDllInHTML(outputDir, defaultAppHtml, log, rebuild) {
   try {
     const htmlTemplate = path.join(outputDir, 'public', 'index.html');
-    let htmlContent;
-    let $;
+    const cssFile = path.join(outputDir, 'public', 'css', 'vendor.css');
 
-    // Check if index.html template exists
-    if (!fs.existsSync(htmlTemplate)){
-      // Read html content from default app index.html
-      htmlContent = fs.readFileSync(defaultAppHtml, 'utf-8');
-      $ = cheerio.load(htmlContent);
-      fs.writeFileSync(htmlTemplate, $.root().html());
-    } else {
-      // Read html content from index.html template
-      htmlContent = fs.readFileSync(htmlTemplate, 'utf-8');
-      $ = cheerio.load(htmlContent);
+    // Read html content from default app index.html
+    const htmlContent = fs.readFileSync(defaultAppHtml, 'utf-8');
+    const $ = cheerio.load(htmlContent);
+
+    // add vendor.css
+    if (fs.existsSync(cssFile) && !rebuild) {
+      $('head').append('<link href="css/vendor.css" rel="stylesheet">');
     }
 
     // Check if vendor.js is included inside the HTML file
-    if ($('script').length === 1 && $('script')[0].attribs.src === "vendor.js") {
-      return;
+    const hasVendor = (Array.from($('script')) || []).some(script => script.attribs.src === 'vendor.dll.js');
+    if (!hasVendor) {
+      $('body').append('<script data-id="dll" src="vendor.dll.js"></script>');
```

Comment:
> src="vendor.dll.js"

Should this be `vendor.js` instead?

Note to self: Added `data-id="dll"` to differentiate between scripts added by the DllPlugin and scripts added by default to the index.html file.

Reply:
If we want to use `vendor.dll.js`, we need to specify something like this in the webpack config:
```
   output: {
      filename: '[name].dll.js',
      ...
   }
```